lib/Doctrine/ORM/LazyCriteriaCollection.php

```diff
@@ -87,21 +79,15 @@ public function contains($element)
         return $this->entityPersister->exists($element, $this->criteria);
     }
 
-    /**
-     * {@inheritDoc}
-     */
-    public function matching(Criteria $criteria)
+    public function matching(Criteria $criteria): ReadableCollection&Selectable
```

Comment:
[Maybe we should switch to Psalm v5 because of this](https://github.com/vimeo/psalm/issues/8229)